File: src/C3.Catalogue/Tapes/ITapeRepository.vb
Namespace Tapes

    Public Interface ITapeRepository

        Function GetAll() As IList(Of Tape)
        Function FindByShortIdentifier(identifier As String) As Tape
        Function ModelExists(identifier As String) As Boolean
        Function NextNumberForModel(identifier As String) As Integer
        Function IdentifierExists(identifier As String, shortIdentifier As String) As Boolean
        Sub AddRange(values As IList(Of Tape))
        Sub Update(value As Tape)
        Sub Delete(shortIdentifier As String)

    End Interface

End Namespace

File: src/C3.Catalogue/Tapes/ITapeRepository.cs
using System.Collections.Generic;

namespace C3.Catalogue.Tapes
{
    public interface ITapeRepository
    {
        IList<Tape> GetAll();

        Tape FindByShortIdentifier(string identifier);

        bool ModelExists(string identifier);

        int NextNumberForModel(string identifier);

        bool IdentifierExists(string identifier, string shortIdentifier);

        void AddRange(IList<Tape> values);

        void Update(Tape value);

        void Delete(string shortIdentifier);
    }
}
